jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -489,7 +491,8 @@ def _plot_uncertainties(self):
     def set_plot_axes(self):
         # Set axes labels for the spectrum viewer
         flux_unit_type = "Flux density"
-        x_unit = u.Unit(self.state.x_display_unit)
+        x_disp_unit = self.state.x_display_unit
+        x_unit = u.Unit(x_disp_unit) if x_disp_unit is not None else u.dimensionless_unscaled
```

Comment:
Empty string should also resolve to `u.dimensionless_unscaled`
```suggestion
        x_unit = u.Unit(x_disp_unit) if x_disp_unit else u.dimensionless_unscaled
```